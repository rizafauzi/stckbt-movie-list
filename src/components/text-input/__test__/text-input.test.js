

import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import TextInput from '../text-input.component'
import { render, cleanup, fireEvent } from '@testing-library/react'


afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement('div')
  ReactDOM.render(<TextInput />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('it has a placeholder prop', () => {
  const onChangeTextMock = jest.fn()
  const { getByTestId } = render(
    <TextInput onChange={onChangeTextMock} placeholder="test-placeholder" />,
  )
  const input = getByTestId('text-input')
  expect(input).toHaveProperty('placeholder')
})


it('it matches snapshot', () => {
  const tree = renderer.create(<TextInput />).toJSON()
  expect(tree).toMatchSnapshot()
})