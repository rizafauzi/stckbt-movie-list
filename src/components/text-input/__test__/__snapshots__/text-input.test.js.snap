// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<input
  className="sc-bdnylx jWnVCv"
  data-testid="text-input"
  placeholder="test"
/>
`;
